code/modules/clothing/masks/transmog.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/clothing/mask/morphing
+	name = "curious mask"
+	desc = "It doesn't really resemble anything, though it gives you an eerie and intense feeling nonetheless."
+	icon_state = "morphing_mask"
+	item_state = "chapmask"
+	flags = FPRINT
+	body_parts_covered = FACE
+	w_class = W_CLASS_SMALL
+	var/target_type = null
+
+/obj/item/clothing/mask/morphing/equipped(M as mob, wear_mask)
+	if(target_type)
+		var/mob/living/carbon/C = M
+		if(!istype(C))
+			return
+		if(C.wear_mask == src)
+			if(target_type != C.type)
+				C.visible_message("<span class='danger'>As [C] puts on \the [src], \his body begins to shift and contort!</span>","<span class='danger'>As you put on \the [src], your body begins to shift and contort!</span>")
+				C.transmogrify(target_type, TRUE)
+
+/obj/item/clothing/mask/morphing/attackby(obj/item/weapon/W, mob/user)
+	if(!target_type)
+		if(istype(W, /obj/item/stack/sheet/animalhide) || istype(W, /obj/item/asteroid/goliath_hide) || istype(W, /obj/item/clothing/head/bearpelt/real))
+			var/obj/item/clothing/mask/morphing/T
+			if(istype(W, /obj/item/asteroid/goliath_hide))
+				T = new /obj/item/clothing/mask/morphing/goliath(get_turf(src))
+			else if(istype(W, /obj/item/clothing/head/bearpelt/real))
+				T = new /obj/item/clothing/mask/morphing/bear(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/corgi))
+				T = new /obj/item/clothing/mask/morphing/corgi(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/cat))
+				T = new /obj/item/clothing/mask/morphing/cat(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/monkey))
+				T = new /obj/item/clothing/mask/morphing/monkey(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/lizard))
+				T = new /obj/item/clothing/mask/morphing/lizard(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/xeno))
+				T = new /obj/item/clothing/mask/morphing/xeno(get_turf(src))
+			else if(istype(W, /obj/item/stack/sheet/animalhide/human))
+				T = new /obj/item/clothing/mask/morphing/human(get_turf(src))
+			if(T)
+				to_chat(user, "<span class='notice'>You wrap \the [W] around \the [src].</span>")
+				if(istype(W, /obj/item/stack/sheet/animalhide))
+					var/obj/item/stack/sheet/animalhide/A = W
+					A.use(1)
+				else
+					qdel(W)
+				if(loc == user)
+					user.drop_item(src, force_drop = 1)
```

Comment:
Yeah but if I `qdel()` it after attempting to put the new mask in the user's hands, the user's hands may still be full. If I `qdel()` it before attempting to put the new mask in the user's hands, I can't measure the `loc` of the old mask, so I don't know whether I need to put the new mask in the user's hands.

The only alternative to have the same functionality would be having two `qdel()`s, one inside the `if()` and one in an else, or to store the `loc`, then `qdel()`, then check whether the `loc` is the user.